uhabits-android/src/main/java/org/isoron/uhabits/activities/common/dialogs/MultipleDialogsHandler.kt

```diff
@@ -0,0 +1,15 @@
+package org.isoron.uhabits.activities.common.dialogs
+
+import android.app.Dialog
+import java.lang.ref.WeakReference
+
+class MultipleDialogsHandler {
+    companion object {
+        var currentDialog: WeakReference<Dialog> = WeakReference(null)
+
+        fun Dialog.dismissCurrent() {
+            currentDialog.get()?.dismiss()
+            currentDialog = WeakReference(this)
+        }
```

Comment:
Splitting my suggested `dismissCurrentAndShow` function into multiple functions is fine, but it's unexpected to me that `dismissCurrent` also sets the reference. I think `dismissCurrent` should just dismiss and (possibly) set the current reference to null. It would be more clear to have a separate `Dialog.setCurrent()` function, or a combined `showAndSetCurrent()`.